[in-proc8] Update deps json test for multi-target

liliankasem opened the pull request:
<!-- Please provide all the information below.  -->

### Issue describing the changes in this PR

#10044

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

<!-- Optional: delete if not applicable  -->
### Additional information

Additional PR information

liliankasem committed:
Update deps json test for multi-target

liliankasem committed:
Update test

liliankasem committed:
Update WebJobs.Script.Tests.csproj

jviau commented on test/WebJobs.Script.Tests/WebJobs.Script.Tests.csproj:
```diff
@@ -116,7 +116,10 @@
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
-    <None Update="Microsoft.Azure.WebJobs.Script.WebHost.deps.json">
+    <None Update="DepsFiles\net8.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json">
```
Instead of needing to know which one to use in the test, you could leverage msbuild to do that for you. Note in this example I am renaming the deps files to include TFM in their name, but on copy I transform back to the original name.

``` xml
<None Update="Microsoft.Azure.WebJobs.Script.WebHost.Net6.deps.json"
    CopyToOutputDirectory="Always"
    TargetPath="Microsoft.Azure.WebJobs.Script.WebHost.deps.json"
    Condition="'%(TargetFramework)' == 'net6.0" />
<None Update="Microsoft.Azure.WebJobs.Script.WebHost.Net8.deps.json"
    CopyToOutputDirectory="Always"
    TargetPath="Microsoft.Azure.WebJobs.Script.WebHost.deps.json"
    Condition="'%(TargetFramework)' == 'net8.0" />
```

jviau replied:
Could simplify to 
```suggestion
    <None Update="DepsFiles\*\Microsoft.Azure.WebJobs.Script.WebHost.deps.json" CopyToOutputDirectory="Always" />
```

And then remove the other lines.

jviau reviewed (commented):
Do we plan on multi-targeting our test projects also? Because that would change how this can be approached.

liliankasem commented:
> Do we plan on multi-targeting our test projects also? Because that would change how this can be approached.

From what've seen, @fabiocav opted not to multi-target. I assume that was intentional

liliankasem committed:
Update test/WebJobs.Script.Tests/WebJobs.Script.Tests.csproj

Co-authored-by: Jacob Viau <javia@microsoft.com>

liliankasem committed:
Update WebJobs.Script.Tests.csproj

liliankasem committed:
Update WebJobs.Script.Tests.csproj

liliankasem commented:
deps file for .net6 stays the same. Here are the change s for net8

Removed:
    - Grpc.AspNetCore.Server.ClientFactory.dll: 2.0.0.0/2.55.0.0
    - Grpc.AspNetCore.Server.dll: 2.0.0.0/2.55.0.0
    - Grpc.Net.Client.dll: 2.0.0.0/2.55.0.0
    - Grpc.Net.ClientFactory.dll: 2.0.0.0/2.55.0.0
    - Grpc.Net.Common.dll: 2.0.0.0/2.55.0.0
    - System.Diagnostics.DiagnosticSource.dll: 6.0.0.0/6.0.1523.11507
    - System.IO.Hashing.dll: 7.0.0.0/7.0.22.51805
    - System.Text.Json.dll: 6.0.0.0/6.0.2523.51912
    - Yarp.ReverseProxy.dll: 2.0.1.0/2.0.123.32202

  Added:
    - Grpc.AspNetCore.Server.ClientFactory.dll: 2.0.0.0/2.55.0.0
    - Grpc.AspNetCore.Server.dll: 2.0.0.0/2.55.0.0
    - Grpc.Net.Client.dll: 2.0.0.0/2.55.0.0
    - Grpc.Net.ClientFactory.dll: 2.0.0.0/2.55.0.0
    - Grpc.Net.Common.dll: 2.0.0.0/2.55.0.0
    - System.IO.Hashing.dll: 7.0.0.0/7.0.22.51805
    - Yarp.ReverseProxy.dll: 2.0.1.0/2.0.123.32202

liliankasem committed:
Remove old deps file

fabiocav reviewed (approved):
Let's get this in. 

This will be reviewed again prior to final merge and the dependencies will also change as we move the references prior to the `runtimeassemblies` update.
